dials.go

```diff
@@ -283,12 +336,50 @@ func (d *Dials) updateSourceValue(
 	}
 }
 
+func (d *Dials) markSourceDone(
+	ctx context.Context,
+	sourceValues []sourceValue,
+	watchTab *watcherDone,
+) bool {
+	// Set the calling source's watching bit to false
+	for i, sv := range sourceValues {
+		if watchTab.source == sv.source {
+			sourceValues[i].watching = false
+			break
+		}
+	}
+
+	// check whether any sources have watching set to true
+	// (using a loop here because it's not worth maintaining an extra
+	// datastructure for an infrequent operation)
+	for _, sv := range sourceValues {
+		if sv.watching {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *Dials) submitCallback(ctx context.Context, ev userCallbackEvent) {
```

Comment:
Hmm -- we're really submitting the _event_ rather than the callback.  The callback manager then takes care of invoking the callback.  Maybe this should be called `submitEvent` or `fireEvent` or something like that instead?